src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -0,0 +1,178 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { fireEvent } from "../../../common/dom/fire_event.js";
+
+import "../../../components/ha-card.js";
+
+import toggleEntity from "../common/entity/toggle-entity.js";
+import isValidEntityId from "../../../common/entity/valid_entity_id.js";
+import stateIcon from "../../../common/entity/state_icon.js";
+import computeStateDomain from "../../../common/entity/compute_state_domain.js";
+import computeStateName from "../../../common/entity/compute_state_name.js";
+import { HomeAssistant } from "../../../types.js";
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+interface Config extends LovelaceConfig {
+  entity: string;
+  name?: string;
+  icon?: string;
+  tap_action?: "toggle" | "call-service" | "more-info";
+  service?: string;
+  service_data?: object;
+}
+
+class HuiEntityButtonCard extends HassLocalizeLitMixin(LitElement) 
+implements LovelaceCard {
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {}
+    };
+  }
+
+  protected hass!: HomeAssistant;
```

Comment:
are the `!` necessary? Where would TS get the idea that it could be null?

Reply:
Not surr. But it shows as an "error" in VS Code. In the Glance card as well. Its not necessary to run so I can take it out

Reply:
Not surr. But it shows as an "error" in VS Code. In the Glance card as well. Its not necessary to run so I can take it out

Reply:
Oh snap you're right. 

We should fail the build if there are errors

Reply:
Opened https://github.com/home-assistant/home-assistant-polymer/pull/1780

Reply:
There is also the error in the glance card for call-service. config.service could be undefined. Line 179 I believe. But I agree on failing if TS errors

Reply:
So set them to `?` instead of `!`. Then use a `!` in each case where you know that it's not empty, or do a check if it exists before using.